crits/standards/parsers.py

```diff
@@ -160,45 +253,179 @@ def parse_observables(self, observables):
+                                               user=analyst,
+                                               description=description,
+                                               title=title,
+                                               data_type="Text",
+                                               tool_name="STIX",
+                                               tool_version=None,
+                                               method=self.source_instance.method,
+                                               reference=self.source_instance.reference)
+                    self.parse_res(imp_type, obs, res)
+                elif (isinstance(item, File) and
+                      item.custom_properties and
+                      item.custom_properties[0].name == "crits_type" and
+                      item.custom_properties[0]._value == "Certificate"):
+                    imp_type = "Certificate"
+                    description = str(item.description)
+                    filename = str(item.file_name)
+                    data = None
+                    for obj in item.parent.related_objects:
+                        if isinstance(obj.properties, Artifact):
+                            data = base64.b64decode(obj.properties.data)
+                    res = handle_cert_file(filename,
+                                           data,
+                                           self.source,
+                                           user=analyst,
+                                           description=description)
+                    self.parse_res(imp_type, obs, res)
+                elif isinstance(item, File) and self.has_network_artifact(item):
+                    imp_type = "PCAP"
+                    description = str(item.description)
+                    filename = str(item.file_name)
+                    data = None
+                    for obj in item.parent.related_objects:
+                        if (isinstance(obj.properties, Artifact) and
+                            obj.properties.type_ == Artifact.TYPE_NETWORK):
+                            data = base64.b64decode(obj.properties.data)
+                    res = handle_pcap_file(filename,
+                                           data,
+                                           self.source,
+                                           user=analyst,
+                                           description=description)
+                    self.parse_res(imp_type, obs, res)
+                elif isinstance(item, File):
+                    imp_type = "Sample"
+                    filename = str(item.file_name)
+                    md5 = item.md5
+                    data = None
+                    for obj in item.parent.related_objects:
+                        if (isinstance(obj.properties, Artifact) and
+                            obj.properties.type_ == Artifact.TYPE_FILE):
+                            data = base64.b64decode(obj.properties.data)
```

Comment:
I was seeing a `File: Incorrect padding` failure message, but only when receiving certain Samples via TAXII Agent. I eventually realized that even the Samples that were making it through without an error had bad data. After some work, I managed to trace this problem to the above line.  I didn't dig very deep into the python-stix code, but it appears that `STIXPackage.from_xml()` does the work of base64 decoding the Sample data. So, when the above line executes, it is trying to decode already decoded data, which causes the error.
Changing `data = base64.b64decode(obj.properties.data)` to `data = obj.properties.data` seems to solve the problem.